blocks/stackcomponent/stackcomponent.js

```diff
@@ -0,0 +1,43 @@
+function decorateStackComponentContent(div) {
+  const content = div;
+  if (content) {
+    // first child
+    const eyeBrow = content.children[0];
+    if (eyeBrow) {
+      eyeBrow.classList.add('qui-stack-component-eyebrow');
+      eyeBrow.classList.add('eyebrow-2');
+      eyeBrow.classList.add('color-accent-tertiary');
+    }
+    const title = content.children[1];
+    if (title) {
+      title.classList.add('qui-stack-component-title');
+    }
+    const description = content.children[2];
+    if (description) {
+      description.classList.add('qui-stack-component-description');
+      description.classList.add('body-1');
+    }
+  }
+}
+
+export default function decorate(block) {
+  const rows = [...block.children];
+  block.classList.add(`stackcomponent-${rows.length}-rows`);
+
+  rows.forEach((row, index) => {
+    row.classList.add('stackcomponent-qui-stack-component');
+    row.classList.add('stackcomponent-row');
+    row.classList.add(`stackcomponent-row-${index + 1}`);
+    const columns = [...row.children];
+    columns.forEach((column) => {
+      // create a wrapper div and move the column inside it
+      const wrapper = document.createElement('div');
+      wrapper.classList.add('stackcomponent-wrapper');
```

Comment:
The naming is a bit unfortunate, because it clashes with the OOTB block wrapper class.
If you look in the DOM, you have two divs with the same class, which means the same CSS is applied twice and could bring unintended side-effects. Is it intentional like this?

<img width="834" alt="Screenshot 2024-09-24 at 13 08 02" src="https://github.com/user-attachments/assets/c7602e6d-232e-46e8-a07f-c95ce001818d">
.

Reply:
fixed